hugr-py/src/hugr/hugr/base.py

```diff
@@ -344,16 +345,20 @@ def _add_node(
         op: Op,
         parent: ToNode | None = None,
         num_outs: int | None = None,
-        metadata: dict[str, Any] | None = None,
+        metadata: dict[str, Any] | NodeMetadata | None = None,
     ) -> Node:
         parent = parent.to_node() if parent else None
-        node_data = NodeData(op, parent, metadata=metadata or {})
+
+        if metadata is None or isinstance(metadata, dict):
+            metadata = NodeMetadata(metadata)
+
+        node_data = NodeData(op, parent, metadata=metadata)
```

Comment:
How is the metadata of `NodeData` connected to that of `Node`? What is the difference?